src/server/gc_worker/compaction_filter.rs

```diff
@@ -37,20 +37,41 @@ lazy_static! {
     static ref GC_CONTEXT: Mutex<Option<GcContext>> = Mutex::new(None);
 }
 
-pub fn init_compaction_filter(
-    db: RocksEngine,
-    safe_point: Arc<AtomicU64>,
-    cfg_tracker: GcWorkerConfigManager,
-    cluster_version: ClusterVersion,
-) {
-    info!("initialize GC context for compaction filter");
-    let mut gc_context = GC_CONTEXT.lock().unwrap();
-    *gc_context = Some(GcContext {
-        db: db.as_inner().clone(),
-        safe_point,
-        cfg_tracker,
-        cluster_version,
-    });
+pub trait CompactionFilterInitializer {
+    fn init_compaction_filter(
+        &self,
+        safe_point: Arc<AtomicU64>,
+        cfg_tracker: GcWorkerConfigManager,
+        cluster_version: ClusterVersion,
+    );
+}
+
+impl<T> CompactionFilterInitializer for T {
+    default fn init_compaction_filter(
+        &self,
+        _safe_point: Arc<AtomicU64>,
+        _cfg_tracker: GcWorkerConfigManager,
+        _cluster_version: ClusterVersion,
+    ) {
```

Comment:
Panic here so we won't forget to implement this method for future implementors?

Reply:
Ah I got it.

Reply:
I don' think so, it's a feature of RocksDB.